core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -359,13 +400,15 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
       throw new IllegalStateException("Controller to broker state change requests batch is not empty while creating a " +
         s"new one. Some UpdateMetadata state changes to brokers $updateMetadataRequestBrokerSet with partition info " +
         s"$updateMetadataRequestPartitionInfoMap might be lost ")
+    metadataInstance = metadataProvider()
```

Comment:
For ZK, the provider always just returns the (one and only) ControllerContext, right?

Reply:
Yes.